### DISCUSSION TITLE
Extracting transmissivity, storage and leakage factor from HantushWellModel parameters

### DISCUSSION THREAD

**Discussion by HenrikSpa** [pastas/pastas#694] [Q&A] [answered]
Hi!
In short, I want as the title says, attain the values of transmissivity, storage and leakage factor from HantushWellModel parameters.

There used to be documentation about this, for example [response_functions version 0.19](https://github.com/pastas/pastas/blob/v0.19.0/concepts/response_functions.ipynb), but I can't find this information in the current version of the documentation. Why was it removed?

### Transmissivity

According to the documentation (the link above), T (transmissivity) can be attained from the parameter A. 

(1)    A = 1 / (4πT)

But the unit for T seems to be directly related to the unit of the well series. If I use a well series with unit L/s (_liters per second_), the parameter A needs to be multiplied with 1000 to get T with unit m^2/s. 

**Question 1:** Is this correct?

### Storage
The storage (S) is somewhat more difficult. According to the link above (documentation for 0.19):

(2)    a = cS 
    and 
(3)    λ^2=cT

With this information, it's possible to attain c from the eq 3 (using T attained from eq 1) and then S from eq 2.

However! The value of S is way to small! But, and this is the interesting part, if I multiply S with 86400 (number of seconds in a day), I get exactly the same S that I attain if manually fit the simulated drawdown curve against Hantush/Thies type curves!

**Question 2a:** Is this extraction of S the correct way to do it?
**Question 2b:** Why do I have to multiply S with 86400? Has it something to do with my unit for the well series (L/s) or that Pastas is somehow using daily timesteps as a base line for the calculations?

### Leakage (leakance) factor K'/b'

Comparing the Hantush equation as you (the Pastas team) wrote in the documentation for 0.19 (the link above) with the original article (_Hantush, M.S., Jacob, C.E. (1955). ”Non-steady radial flow in an infinite leaky aquifer”_), it gives that what Pastas labels as  λ^2 is labeled as B^2 by Hantush.

Hantush (1955) then describe B^2 = T / (K'/b'), after which it's possible to attain K'/b'.

The value for K'/b' seem to be correct (after multiplying A with 1000 using flow with unit L/s) when I fit the simulated drawdown curve manually against Hantush/Thies type curves.

**Question 3:** Is extracting K'/b' this way, the correct way to do it?

### Followups

This is more of a statement, but simulating HantushWellModel on daily or hourly timesteps, using well series with the same unit (L/s) gives the same draw down after the same amount of time. It seems like this would cause the extraction of transmissivity, storage and leackage to be independent of the timestep of the simulatation (which is a good thing!).

Regards,
Henrik

**Comment by mbakker7** [comment]
Thanks for your question, @HenrikSpa.

You are on the right track.

Question 1

The units of A are m / (units of the stress). So in your case meters / (Liters / sec), so that the units of T are L / (meter sec) = 10^(-3) m^2/s.

Question 2

Not sure what you are trying to do with equation 3. Shouldn't you use c = r^2 / (4bT)?

Question 3

Why are you not using
K'/b' = 1 / c

I guess I answered Question 1. For Q2 and Q3 I await your feedback to the questions I posed. Hope this helps.

**Reply by mbakker7** [reply]
The notebook you referred to is now at https://pastas.readthedocs.io/en/master/benchmarks/check_response_functions.html
But I just noticed that the equations don't render for some reason. Can you make an Issue for that?

**Comment by HenrikSpa** [comment]
Thanks a lot for the answers!

> Thanks for your question, @HenrikSpa.
> 
> You are on the right track.
> 
> Question 1
> 
> The units of A are m / (units of the stress). So in your case meters / (Liters / sec), so that the units of T are L / (meter sec) = 10^(-3) m^2/s.

That explains a lot, thanks!

> Question 2
> 
> Not sure what you are trying to do with equation 3. Shouldn't you use c = r^2 / (4bT)?

The equations did work using this link ([check_response_functions 1.4.0](https://github.com/pastas/pastas/blob/v1.4.0/doc/benchmarks/check_response_functions.ipynb)) (which I found from your answer)

Your suggestion is the way I extract c, I should have made it more clear. Here is the full method for attaining S:

In the new link it says that (for HantushWellModel)

b = ln(1/4λ^2) 

And substituting λ^2 for cT (eq 2) gives

b = ln(1/4cT) 

which gives c as

c = 1/ 4exp(b)T

And then attaining S from eq 2 as

S = a/c

So to make my question 2a more clear:

The equation  λ^2 = cT was not written among the equations for HantushWellModel, so I was unsure if it was correct to use that relation. I was also unsure because I got S values that was 86400 times to small. 

But from your answer I understand that it's correct to use  λ^2 = cT for HantushWellModel as well.

Maybe λ^2=cT should be written among the equations for HantushWellModel?

> Question 3
> 
> Why are you not using K'/b' = 1 / c

You are correct and that's the way I do it, I should have written all steps for this question also:

Knowing B^2 = T / (K'/b'),   B^2 = λ^2 and λ^2=cT then:

λ^2 = T / (K'/b') becomes:

cT = T / (K'/b') which gives, as you wrote

K'/b' = 1 / c

So for question 3 I guess I was just looking for confirmation that this is the correct way to do it (that B^2 = λ^2 and λ^2 = cT) , which from your answer I suppose it is.

> I guess I answered Question 1. For Q2 and Q3 I await your feedback to the questions I posed. Hope this helps.

It helps a lot! Thanks!

So I think the only thing I don't get now is question 2b:

**Question 2b:** Why do I have to multiply S with 86400? Has it something to do with my unit for the well series (L/s) or that Pastas is somehow using daily timesteps as a base line for the calculations?

Regards,
Henrik

**Comment by mbakker7** [comment, marked as answer]
@HenrikSpa, 

Computations get a bit confusing as the parameter b' is log(b). This log scaling is needed during the optimization process, to make sure b isn't negative during part of the iterations.

Anyway, when computing c = 1 / (4Texp(b')), you get c in the same time units as used in T. So if T is m/d, then c is in days. If T is in m/s, then T is in seconds. You have to convert c to days for the final computation.

When computing S = a / c, then a is units of days (the default in Pastas and I assume you are using) and if you converted c to days in the step above, then you are good to go.

Hope this helps,

Mark

**Reply by HenrikSpa** [reply]
This explains a lot! Thank you for the answers!

Regards,
Henrik
